thunder/tests/test_auto_register_torchops.py

```diff
@@ -0,0 +1,187 @@
+
+    import thunder.torch.default_torch_ops as ops
+    from torch.testing._internal.common_methods_invocations import op_db
+    from torch.testing._internal.common_device_type import skipCPUIfNoLapack
+
+    name2func = {}
+    [name2func.setdefault(v.__name__, v) for v in ops.torch_auto_registered_ops[torch]]
+    [name2func.setdefault(f"nn.functional.{v.__name__}", v) for v in ops.torch_auto_registered_ops[torch.nn.functional]]
+    # Use the sample input from torch.xx to test torch.tensor.xx
+    [name2func.setdefault(f"Tensor.{v.__name__}", v) for v in ops.torch_auto_registered_ops[torch.Tensor]]
+    op_infos = [opinfo for opinfo in op_db if opinfo.name in name2func]
+    total = 0
+    cnt = 0
+    suc = 0
+    for op_info in op_infos:
+        if device == "cuda" and torch.float32 not in op_info.dtypesIfCUDA:
+            continue
+        if device == "cpu" and not torch.float32 in op_info.dtypes:
+            continue
+        # No cuda backend support
+        if op_info.name in ("nonzero_static",) and device == "cuda":
+            continue
+        if device == "cpu" and not torch._C.has_lapack and skipCPUIfNoLapack in op_info.decorators:
+            continue
+        funcs = [name2func[op_info.name], name2func.get(f"Tensor.{op_info.name}", None)]
+        for func in funcs:
+            if func is None:
+                continue
+            total += 1
+            # It takes too long, test only the first 5 sample inputs
+            gen = islice(
+                op_info.sample_inputs_func(
+                    op_info, device=torch.device(device), dtype=torch.float32, requires_grad=requires_grad
+                ),
+                5,
+            )
+            for sample in gen:
+                try:
+                    jfun = thunder.jit(func)
+                    out = jfun(sample.input, *sample.args, **sample.kwargs)
+                except Exception as e:
+                    cnt += 1
+                    assert isinstance(e, NotImplementedError)
+                    assert str(e).startswith(f"Exception encountered when doing automatic registration") or str(
+                        e
+                    ).startswith(f"Unsupported type:")
+                    break
+                else:
+                    if requires_grad:
+                        trc = thunder.last_backward_traces(jfun)[-1]
+                        fwd_trc = thunder.last_traces(jfun)[-1]
+                        # skip if it is not differentiable
+                        outs = fwd_trc.output[0]["output"]
+                        outs = outs if isinstance(outs, tuple) else (outs,)
+                        if all(not thunder.core.dtypes.is_inexact_dtype(o.dtype) for o in outs):
+                            continue
+                        vjp_op_name = f"{op_info.name.split('.')[-1]}_vjp"
+                        if op_info.name == "mm":
+                            assert any(bsym.sym.name.endswith(vjp_op_name) for bsym in trc.bound_symbols)
```

Comment:
could you spell out the name of `bsym.sym`, instead of `.endswith(vjp_op_name)`?
as it's purely a special case for "mm", I'd say it's okay to hard code

Reply:
It's a bit tricky here, for testing `torch.Tensor.mm` and `torch.mm`, both of their op_info.name is `mm`, and because the `torch.mm` and `torch.spmm` are aliases, they are actually the same callable object, so in the `_torch_to_thunder_function_map` they map to the same symbol name `torch.spmm`. So here for `torch.Tensor.mm` the backward symbol name is `mm_vjp` and for `torch.mm` is the `spmm_vjp`, so it's checked with endswith(`mm_vjp`)
Currently we do not distinguish alias, the ops are just listed in the file alphabetically, so the last one that appeared is the name shown in the trace, I think it does not matter which name is used since they are alias.